explainers/signed-exchange-alternate-link-explainer.md

```diff
@@ -0,0 +1,164 @@
+      ```
+      <link href="https://feed.example/article.html.sxg"
+            rel="prefetch">
+      <a href="https://feed.example/article.html.sxg">feed.example</a>
+      ```
+   - While the user of the UA is browsing an article (article.html), and if there is a signed exchange alternate link, the UA can provide the user with a way to save the page in signe exchange format. The saved file can be used to share with other users.
+
+### Recursive subresource signed exchange prefetch
+1. A crawler of an aggregator site fetches an article (https://publisher.example/article.html).
+1. There is an alternate link of signed exchange of the article in the response header:
+    ```
+    Link: <https://sxg.publisher.example/article.html.sxg>;
+            rel="alternate";
+            type="application/signed-exchange;v=b3";
+            anchor="https://publisher.example/article.html"
+    ```
+1. The crawler fetches and verifies the signed exchange (article.html.sxg).
+
+   1. The response from the server has an alternate link of subresource signed exchange:
+      ```
+      Link: <https://sxg.publisher.example/lib.js.sxg>;
+              rel="alternate";
+              type="application/signed-exchange;v=b3";
+              anchor="https://cdn.publisher.example/lib.js"
+      ```
+   1. And the inner response of the main resource signed exchange (article.html.sxg) has a preload header and an [allowed-alt-sxg header](./signed-exchange-subresource-subtitution-explainer.md):
+      ```
+      Link: <https://cdn.publisher.example/lib.js>;
+              rel="preload";
+              as="script"
+      Link: <https://cdn.publisher.example/lib.js>;
+              rel="allowed-alt-sxg";
+              header-integrity="sha256-XXXXXX"
+      ```
+1. The crawler fetches and verifies the subresource signed exchange (lib.js.sxg).
+1. The aggregator site will serve the signed exchanges (article.html.sxg and lib.js.sxg) from their own server.
+
+1. While a user is browsing the aggregator site (https://feed.example), a prefetch link element is inserted because the use is likely to want to read the article.
+   ```
+   <link rel="prefetch" href="https://feed.example/sxg.publisher.example/article.html.sxg">
+   ```
+1. The UA prefetches the signed exchange (article.html.sxg).
+   1. The response from the server has an alternate link of subresource signed exchange:
+      ```
+      Link: <https://feed.example/sxg.publisher.example/lib.js.sxg>;
+              rel="alternate";
+              type="application/signed-exchange;v=b3";
+              anchor="https://cdn.publisher.example/lib.js"
+      ```
+   1. And the inner response of the main resource signed exchange (article.html.sxg) has a preload header and an allowed-alt-sxg header:
```

Comment:
```suggestion
To prevent an attacker from loading an incompatible version of the subresource, the resource _inside_ the signed exchange has to identify the exact version of the replacement signed exchange using a `Link: ... rel="allowed-alt-sxg"` with the hash of the signed headers (which themselves include a hash of the content).

To prevent a tracker from conveying a user ID in their choice of which subresources to prefetch, the inner resource has to preload the same subresources that the aggregator prefetches.
```